packages/core-mobile/app/screens/send/utils/avm/validate.ts

```diff
@@ -0,0 +1,45 @@
+import { Avalanche } from '@avalabs/core-wallets-sdk'
+import { TokenWithBalanceAVM } from '@avalabs/vm-module-types'
+import { Amount } from 'types'
+import { bigIntToString } from '@avalabs/core-utils-sdk'
+import { SendErrorMessage } from '../types'
+import { getAvalancheSendMaxAmount } from '../getAvalancheSendMaxAmount'
+
+export const validate = ({
+  amount,
+  address,
+  maxFee,
+  token,
+  onCalculateMaxAmount
+}: {
+  amount: bigint
+  address: string
+  maxFee: bigint
+  token: TokenWithBalanceAVM
+  onCalculateMaxAmount?: (maxAmount: Amount) => void
+}): void => {
+  const maxAmountValue = getAvalancheSendMaxAmount(token.available, maxFee)
+
+  onCalculateMaxAmount?.({
+    bn: maxAmountValue,
+    amount: maxAmountValue ? bigIntToString(maxAmountValue, token.decimals) : ''
+  })
+
+  if (
+    !Avalanche.isBech32Address(address, false) &&
+    !Avalanche.isBech32Address(address, true)
+  )
+    throw new Error(SendErrorMessage.INVALID_ADDRESS)
+
+  if (!maxFee || maxFee === 0n)
+    throw new Error(SendErrorMessage.INVALID_NETWORK_FEE)
+
+  if (maxAmountValue === 0n)
+    throw new Error(SendErrorMessage.INSUFFICIENT_BALANCE)
```

Comment:
should this be insufficient error or something else?

Reply:
do you have any idea? I think `SendErrorMessage.INSUFFICIENT_BALANCE` make sense because `maxAmountValue` is basically your balance - fee.